contracts/proxy/Clones.sol

```diff
@@ -16,27 +18,34 @@ pragma solidity ^0.8.20;
  */
 library Clones {
     /**
-     * @dev A clone instance deployment failed.
+     * @dev Deploys and returns the address of a clone that mimics the behaviour of `implementation`.
+     *
+     * This function uses the create opcode, which should never revert.
      */
-    error ERC1167FailedCreateClone();
+    function clone(address implementation) internal returns (address instance) {
+        return clone(implementation, 0);
+    }
 
     /**
      * @dev Deploys and returns the address of a clone that mimics the behaviour of `implementation`.
      *
      * This function uses the create opcode, which should never revert.
      */
-    function clone(address implementation) internal returns (address instance) {
+    function clone(address implementation, uint256 value) internal returns (address instance) {
```

Comment:
IMO, that is factory design, which is beyond the scope of this library. Most factory will not use value. Those that do need to take care of that.

IMO it's like Address library (that can do call with value) and TrustedForwarder (that can trigger that call). We don't have any warning in Address about the value issue.

Reply:
Right, indeed, we don't have a warning in Address nor ERC2771Forwarder, but the main difference is the use case:

- ERC2771Forwarder's balance should be always zero since every call forwarded goes along with the value and everything else is refunded
- Address is more generic and it's fine if the caller doesn't have enough balance

Although Clones is very similar to Address, it's more users will need to consider designing their factory in such a way that always has enough balance. Even though that's factory design, I think it make sense to make a simple recmmendation:

```
NOTE: Using a non-zero value at creation will require the contract using this function (e.g. a factory)
to always have enough balance for new deployments. Consider exposing this function under a payable
method.
```

Reply:
That warning is fine

Reply:
Yeah probably true 😂, but ideally we would get more experienced developers like @k06a in the long run.

Think that with the Account Abstraction trend, I'd expect more factories to deploy accounts, and therefore more developers requiring initial value.